phat/tests/legoActions.test.ts

```diff
@@ -200,12 +300,35 @@ describe("Run lego actions", () => {
       const arg_param_0 = Array(20).fill(0);
       // 32 byte `bytes calldata price`, this should be consturcted from the output of last step
 
-      const actions_json = `[
-        {"cmd": "fetch", "config": ${cfg({
-          returnTextBody: true,
-          url: "https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=BTC,USD,EUR",
-        })}},
-        {"cmd": "eval", "config": "Math.round(JSON.parse(input.body).USD)"},
+      const lensApiRequest = {
+        url: 'https://api-mumbai.lens.dev/',
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: toHexString({
+          query: `
+          query Profile {
+            profile(request: { profileId: "0x01" }) {
+              stats {
+                totalFollowers
+                totalFollowing
+                totalPosts
+                totalComments
+                totalMirrors
+                totalPublications
+                totalCollects
+              }
+            }
+          }
+          `,
+        }),
+        returnTextBody: true,
+      };
+
+      const actions_lens_api = `[
+        {"cmd": "fetch", "config": ${cfg(lensApiRequest)}},
+        {"cmd": "eval", "config": "JSON.parse(input.body).data.profile.stats.totalFollowers"},
```

Comment:
Looks a bad idea to write so many `eval`. I'd imagine something like this:

```js
[
  { cmd: fetch: '....' },  // output[0]
  { cmd: eval: config: `
      const followers = JSON.parse(output[0].body).data.profile.stats.totalFollowers;
      const ethAbiEncoded = numToUint8Array32(followers);
      const call = scale.encode(['${arg_to}', [${arg_abi}], '${arg_function}', [[${arg_param_0}], ethAbiEncoded]], scale.encodeBuildTx)
      return call;
   `},
  { cmd: 'call', config: cfg({ callee: calleeEvmTransaction, selector: selectorBuildTransaction }) },
  { cmd: 'eval', config: `
      const decoded = scale.decode(output[2], scale.decodeResultVecU8);
      const walletArgs = scale.encode(decoded.content, scale.encodeVecU8);
      return walletArgs;
    `},
  // ....
]
```

So basically, instead of writing in a unix pipe style (`cmd1 | cmd2 | cmd3`), we can write more complex logic in a native js way easily. Every step can access the output from any previous step referenced by its index.

Reply:
It's totally okey to write complex logic now. The e2e puts it this way to clearly show each transform step.

Also many steps above will be omitted when we enable auto encode/decode. I will update the workflow that time.

Reply:
I strongly against the pipe-style config. You can do it for this case, but it may not be the case in other cases.